homeassistant/components/light/yeelight.py

```diff
@@ -286,13 +327,63 @@ def set_flash(self, flash) -> None:
             except BulbException as ex:
                 _LOGGER.error("Unable to set flash: %s", ex)
 
+    @_cmd
+    def set_effect(self, effect) -> None:
+        """Activate effect."""
+        if effect:
+            from yeelight import (Flow, BulbException)
+            from yeelight.transitions import (disco, temp, strobe, pulse,
+                                              strobe_color, alarm, police,
+                                              police2, christmas, rgb,
+                                              randomloop, slowdown)
+            # transition = int(self.config[CONF_TRANSITION])
+            if effect == EFFECT_DISCO:
+                flow = Flow(count=0, transitions=disco(),)
```

Comment:
Ah i see, That works! thank you i have changed it. And i completely agree with having the effects over there.